paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetViewModel.kt

```diff
@@ -148,8 +150,56 @@ internal class CustomerSheetViewModel(
         scope = viewModelScope.plus(workContext)
     )
 
+    private val isEditing = MutableStateFlow(false)
+    private val selectionConfirmationState = MutableStateFlow(
+        SelectionConfirmationState(
+            isConfirming = false,
+            error = null,
+        )
+    )
+    private val customerState = MutableStateFlow(
+        CustomerState(
+            paymentMethods = listOf(),
+            configuration = configuration,
+            currentSelection = originalPaymentSelection,
+            metadata = null,
+        )
+    )
+
+    private val selectPaymentMethodState = combineAsStateFlow(
+        customerState,
+        selectionConfirmationState,
+        isEditing,
+    ) { customerState, selectionConfirmationState, editing ->
+        val paymentMethods = customerState.paymentMethods
+        val paymentMethodMetadata = customerState.metadata
+        val paymentSelection = customerState.currentSelection
+
+        val userCanEditAndIsEditing = editing && customerState.canEdit
+        val primaryButtonVisible = !userCanEditAndIsEditing && originalPaymentSelection != paymentSelection
+
+        CustomerSheetViewState.SelectPaymentMethod(
+            title = configuration.headerTextForSelectionScreen,
+            savedPaymentMethods = paymentMethods,
+            paymentSelection = paymentSelection,
+            isLiveMode = isLiveModeProvider(),
+            canRemovePaymentMethods = customerState.canRemove,
+            cbcEligibility = customerState.cbcEligibility,
+            primaryButtonVisible = primaryButtonVisible,
+            isGooglePayEnabled = paymentMethodMetadata?.isGooglePayReady == true,
+            isEditing = userCanEditAndIsEditing,
+            isProcessing = selectionConfirmationState.isConfirming,
+            errorMessage = selectionConfirmationState.error,
+            allowsRemovalOfLastSavedPaymentMethod = configuration.allowsRemovalOfLastSavedPaymentMethod,
+            primaryButtonLabel = resources.getString(R.string.stripe_paymentsheet_confirm),
+            mandateText = paymentSelection?.mandateText(
+                merchantName = configuration.merchantDisplayName,
+                isSetupFlow = false,
+            )?.takeIf { primaryButtonVisible }
+        )
+    }
```

Comment:
Moves all decisions for building the `SelectSavedPaymentMethodState` in here. I opted to manage an instance here then will remove when we move to interactors. Need to the `StateFlow` updates first.